### PR TITLE
Fix join part bug

### PR DESCRIPTION
**Pull request by aboft**
- Refactor Users table to hold a role as well as nick
- Alter deleteUser from finding a 'like' user and match exact to prevent crash
- move NAMES command into a function to call
- add feature to auto promote/revoke roles on change

### CODE REVIEW

File: backend/irc/IrcMessageProcessor.ts [hunk 1]
```diff
@@ -10,15 +10,16 @@ import myEmitter from './utils/emitter';
  */
 class IrcMessageProcessor {
   private static _instance: IrcMessageProcessor;
+  private modeState = false;
   private readonly client: TLSSocket;
   private readonly joinConfig: JoinConfig;
-  private readonly names: string[];
+  //private readonly names: string[];
```
Comment (LoLei) [inline backend/irc/IrcMessageProcessor.ts]: Can these commented lines be removed?

File: backend/junit.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    <testcase classname=" warning is printed if number of users differs" name=" warning is printed if number of users differs" time="0.022">
+    </testcase>
+    <testcase classname=" insert and delete nick on join/part" name=" insert and delete nick on join/part" time="0.066">
+    </testcase>
+  </testsuite>
+  <testsuite name="undefined" errors="0" failures="0" skipped="0" timestamp="2020-12-16T15:17:31" time="0.098" tests="1">
+    <testcase classname=" raw IRC message line is parsed to correct IrcMessage object" name=" raw IRC message line is parsed to correct IrcMessage object" time="0.001">
+    </testcase>
+  </testsuite>
+</testsuites>
```
Comment (LoLei) [inline backend/junit.xml]: This file shouldn't be committed. Add it to the gitignore? Or shouldn't it already be in there? Maybe the path is wrong.

File: backend/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "lmrdashboard-backend",
-  "version": "0.3.0",
+  "version": "0.3.1",
```
Comment (LoLei) [inline backend/package.json]: The version upgrade should only happen after it has been merged into the main branch, so we can see if the changes work in the staging environment. If that is the case, the release/version upgrade can be made on the main branch so it's deployed to prod. So I guess you could revert all version changes that happen in this PR?

**Review by LoLei — CHANGES_REQUESTED** [review]

File: backend/irc/IrcMessageProcessor.ts (hunk 1 above)
Comment (aboft) [reply backend/irc/IrcMessageProcessor.ts]: removed

File: backend/junit.xml (hunk 2 above)
Comment (aboft) [reply backend/junit.xml]: removed and added into gitignore

File: backend/package.json (hunk 3 above)
Comment (aboft) [reply backend/package.json]: rolled back version numbers

File: backend/irc/IrcMessageProcessor.ts (hunk 1 above)
Comment (LoLei) [reply backend/irc/IrcMessageProcessor.ts]: They're here (again?)

Comment (aboft) [reply backend/irc/IrcMessageProcessor.ts]: pesky little suckers

**Pull request merged**